opal/util/misc.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024      Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+/** @file:
+ * Miscellaneous utilities
+ */
+
+#ifndef OPAL_UTIL_MISC_H
+#define OPAL_UTIL_MISC_H
+
+#include "opal/include/opal_config.h"
+
+BEGIN_C_DECLS
+
+#ifndef container_of
+#    define container_of(ptr, type, member) ((type *) (((char *) (ptr)) - offsetof(type, member)))
```

Comment:
Should we namespace this to `opal_container_of` so that there is no conflict if another project has the brilliant idea of providing their own copy?

Reply:
Hmmmm container_of is quite common in the wild and we are actually not making a fancier option than the default one, so I'm inclined to keep the name(also we are doing ifndef so it should be safe).

Reply:
I missed the `#ifndef` so I guess it's fine and keeps the impact small.